lib/providers/battery_level_provider.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import 'package:yonomi_device_widgets/providers/power_trait_provider.dart';
+import 'package:yonomi_device_widgets/providers/widget_state.dart';
+import 'package:yonomi_platform_sdk/yonomi-sdk.dart';
+
+class BatteryLevelProvider extends ChangeNotifier {
+  Device? _deviceDetail;
+
+  late String _deviceId;
+
+  late Request _request;
+
+  WidgetState _state = WidgetState.idle;
+
+  String _latestErrorMsg = "An error occurred.";
+
+  BatteryLevelProvider(Request request, String deviceId,
+      {GetDeviceDetailsMethod getDetails =
+          DevicesRepository.getDeviceDetails}) {
+    this._deviceId = deviceId;
+    this._request = request;
+    fetchData(getDeviceDetails: getDetails);
+  }
+
+  Device? get deviceDetail => _deviceDetail;
+
+  Future<Device?> fetchData(
```

Comment:
This probably should be void. I notice same issue with power trait as well provider as well. This method should just set deviceDetail and notify